.github/workflows/release.yml

```diff
@@ -0,0 +1,145 @@
+
+name: Build release
+
+permissions:
+  actions: read
+  checks: read
+  contents: write
```

Comment:
can we add comments for what the different permissions are needed? That be great. eg it's not obvious why "contents" requires "write". This will make it easier in future to maintain it

Reply:
I've updated the permissions. Actually only `contents` is required for creating the tag and the release. added a comment there.